checks/shell_download_validate.go

```diff
@@ -401,7 +400,63 @@ func isPipUnpinnedDownload(cmd []string) bool {
 		}
 	}
 
-	return isInstalled
+	return isInstall
+}
+
+func isPythonCommand(cmd []string) bool {
+	for _, pi := range pythonInterpreters {
+		if isBinaryName(pi, cmd[0]) {
+			return true
+		}
+	}
+	return false
+}
+
+func isPythonPipInstall(cmd []string) bool {
+	if !isPythonCommand(cmd) {
+		return false
+	}
+
+	isInstall := false
+	hasPip := false
+
+	for i := 1; i < len(cmd); i++ {
+		// Search for pip module.
+		if strings.EqualFold(cmd[i], "-m") &&
+			i < len(cmd)-1 &&
+			strings.EqualFold(cmd[i+1], "pip") {
+			hasPip = true
+			continue
+		}
+
+		// Search for install commands.
+		if strings.EqualFold(cmd[i], "install") {
+			isInstall = true
+			continue
+		}
+
+		if isInstall && hasPip {
+			return true
```

Comment:
Maybe just `break` here?

Reply:
i've updated the function as per https://github.com/ossf/scorecard/pull/611/files/5f1fdea08e75ccf719ccb959e40af0c0e16021c5#r658350655, so this function is simpler now. please ptal.